test/WebJobs.Script.Tests/ScriptHostTests.cs

```diff
@@ -337,6 +337,7 @@ public async Task Initialize_WithInvalidSiteExtensionVersion_Throws(string exten
             try
             {
                 string rootPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+                Directory.CreateDirectory(rootPath);
```

Comment:
Looks like some behavior changed that forced you to create this throughout the tests, can you clarify? Were we able to cope with this before but have an issue now?

Reply:
The reason i had to add it is because the tests did not expect the host to read the function metadata before. Because of this it did not matter if the rootpath existed or not. But now that we have moved up the logic to read the metadata the tests invariably do need the path to be present. I was avoiding any change to the read metadata logic of the function. But i had updated it to return an empty collection of function metadata in case the rootpath does not exist.